yw_website/website/views.py

```diff
@@ -1,79 +1,73 @@
+            workflow_list = workflow_list.exclude(pk=workflow.id)
+        else:
+            workflow.graph = latest_version.yw_graph_output
+            workflow.version_id = latest_version.id
+            workflow.version_modified = latest_version.last_modified
+
+    paginator = Paginator(workflow_list, 10)
     page = request.GET.get('page')
-    documents = paginator.get_page(page)
+    workflows = paginator.get_page(page)
     host = request.get_host()
-    return render(request, 'pages/home_page.html', { 'document_list': documents, 'host': host })
 
-@login_required()
+    return render(request, 'pages/home_page.html', { 'workflow_list': workflows, 'host': host  })
+
+@login_required(login_url='/accounts/login/')
 def my_workflows(request):
-    documents_list = Workflow.objects.all().filter(user=request.user)
-    for document in documents_list:
-        latest_version = Version.objects.filter(workflow_id=document.id).order_by('last_modified').first()
-        document.graph = latest_version.yw_graph_output if latest_version is not None else ""
-    
-    paginator = Paginator(documents_list, 10)
+    workflow_list = Workflow.objects.all().filter(user=request.user).exclude(version__isnull=True)
+    for workflow in workflow_list:
+        latest_version = Version.objects.filter(workflow=workflow).order_by('last_modified').first()
+        if latest_version is None:
+            workflow_list = workflow_list.exclude(pk=workflow.id)
+        else:
+            workflow.graph = latest_version.yw_graph_output
+            workflow.version_id = latest_version.id
+
+    paginator = Paginator(workflow_list, 10)
     page = request.GET.get('page')
-    documents = paginator.get_page(page)
+    workflows = paginator.get_page(page)
     host = request.get_host()
-    return render(request, 'pages/my_workflows.html', { 'document_list': documents, 'host': host })
 
-def detailed_workflow(request, document_id):
+    return render(request, 'pages/home_page.html', { 'workflow_list': workflows, 'host': host  })
+def detailed_workflow(request, workflow_id, version_id):
     try:
-        # TODO: change to get object or 404
         if request.method == "GET":
-            document = Workflow.objects.get(pk=document_id)
-            info = {'document': document}
+            form = request.POST
+            workflow = Workflow.objects.get(pk=workflow_id)
+            version = Version.objects.get(pk=version_id)
+            versions = Version.objects.filter(workflow=workflow)
+
+            runs = Run.objects.filter(version=version)
+            info = {'workflow': workflow, 'version': version, 'versions':versions ,'run_list':runs, 'form': form}
             return render(request, 'pages/detailed_workflow.html', info)
-    except Workflow.DoesNotExist:
+        elif request.method == "POST":
+            new_version = request.POST['version_id']
+            return redirect('/detailed_workflow/{}/{}/'.format(workflow_id, new_version))
```

Comment:
I like the way some fo these endpoints are formatted for reference: https://hackernoon.com/restful-api-designing-guidelines-the-best-practices-60e1d954e7c9